### PR TITLE
Setup CI for backend and UI tests [WD-11367]

### PR DESCRIPTION
**Pull request by mas-who**
## Done
 - Added static code analysis for backend
 - Added unit test for backend
 - Added e2e test for UI (using compiled go binary)

## Notes
 - Left out playwright publishing workflow for now since github pages does not work for private repos without a paid plan

### CODE REVIEW

File: ui/blob-report/report.zip
Comment (edlerd) [inline ui/blob-report/report.zip:1]: This should be ignored and not committed, right?

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -21,34 +24,145 @@ jobs:
...
+      - name: Make LXD Site Manager Binaries
+        run: make compile
+
+      - name: Run site manager daemon
+        run: lxd-site-mgrd --state-dir ./state/dir1/ &
+
+      - name: Initialise site manager
+        run: |
+          lxd-site-mgr --state-dir ./state/dir1/ waitready
+          lxd-site-mgr --state-dir ./state/dir1/ init "member1" 0.0.0.0:9001 --bootstrap
```
Comment (edlerd) [inline .github/workflows/pr.yaml:126]: Shall we rely on `make` and then `yarn backend-run` as well as `yarn backend-init` here? Then we would cover checking our dev setup works as described on the side.

File: ui/blob-report/report.zip
Comment (mas-who) [reply ui/blob-report/report.zip:1]: oops that slipped through the cracks

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (mas-who) [reply .github/workflows/pr.yaml:126]: hmm I was hoping to use the CI to test the compiled binaries, so it's almost like testing the snaps, later on we could also potentially compile the go binaries across different os + arch and run those tests in a matrix, but maybe that's too much and we can instead cross check our dev scripts?

Comment (edlerd) [reply .github/workflows/pr.yaml:126]: Right, probably better to run tests closer to the real world than with the dev builds.

Comment (mas-who) [reply .github/workflows/pr.yaml:126]: So keeping it as is?

Comment (edlerd) [reply .github/workflows/pr.yaml:126]: Yeah.

**Review by edlerd — APPROVED** [review]
LGTM

**Pull request merged**